<?xml version="1.0" encoding="utf-8"?>
<launch>

  <!-- 1. To debug just add parameter debut:=true in command window -->
  <arg name="debug" default="false"/>
  <arg name="side"  default="right"/>

  <!-- 1. Load side we want to publish on -->
 <param name="side" type="string" value="$(arg side)" />

  <!-- 3. launch-prefix runs gdb in xterm without running it automatically -->
  <arg     if="$(arg debug)" name="gdb_arg" value="xterm -e gdb --args" />
  <arg unless="$(arg debug)" name="gdb_arg" value="" />

  <!-- 2. Run the setpoint_publisher node -->
  <!-- In isolation:"2 0 1 0 -1 0.2 0.2 0.2 1 0 5 0 0.015 0.015 0.05"-->
  <!-- F alone:"1 0 | 3 0 -7.5 | 0.2 -0.0001 0.1" -->
  <!-- M/F:"2 1 | 0 -10 0 | 0.015 0.015 0.0015 | 0 | 3 0 -7.5 | 0.2 0.0001 0.1-->
  <node name="$(arg side)_set_point_publisher" pkg="force_controller" type="setPoint_pub" output="screen" respawn="true" args="1 0 0 0 0 0.2 -0.0001 0.1" launch-prefix="$(arg gdb_arg)" >  
  </node>

</launch>
